### DISCUSSION TITLE
When I'm trying to build my NextJs project i'm getting an error related to ESLint dependency.

### DISCUSSION THREAD

**Discussion by 2tech-development** [vercel/next.js#51744] [Help]
### Summary

Hi, When I'm trying to build my NextJs project and I'm getting an error related to ESLint dependency. 

I already tried deleting the node_modules and reinstalling it.

### Additional information

```js
Error Message is as follows:

npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: eslint-config-next@12.0.4
npm ERR! Found: eslint@8.33.0
npm ERR! node_modules/eslint
npm ERR!   eslint@"8.33.0" from the root project
npm ERR!   peer eslint@"*" from eslint-import-resolver-typescript@2.7.1
npm ERR!   node_modules/eslint-import-resolver-typescript
npm ERR!     eslint-import-resolver-typescript@"^2.4.0" from eslint-config-next@12.0.4
npm ERR!     node_modules/eslint-config-next
npm ERR!       eslint-config-next@"^12.0.4" from the root project
npm ERR!   5 more (eslint-plugin-import, eslint-plugin-jsx-a11y, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer eslint@"^7.23.0" from eslint-config-next@12.0.4
npm ERR! node_modules/eslint-config-next
npm ERR!   eslint-config-next@"^12.0.4" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: eslint@7.32.0
npm ERR! node_modules/eslint
npm ERR!   peer eslint@"^7.23.0" from eslint-config-next@12.0.4
npm ERR!   node_modules/eslint-config-next
npm ERR!     eslint-config-next@"^12.0.4" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /vercel/.npm/_logs/2023-06-24T05_01_06_036Z-eresolve-report.txt
npm ERR! A complete log of this run can be found in:
npm ERR!     /vercel/.npm/_logs/2023-06-24T05_01_06_036Z-debug-0.log
Error: Command "npm install" exited with 1
BUILD_UTILS_SPAWN_1: Command "npm install" exited with 1
```

**Comment by MrAlaminH** [comment]
The error message is indicating that there is a conflict with the version of eslint that you have installed and the version that eslint-config-next is expecting. 

Your project has eslint version 8.33.0 installed, but eslint-config-next is expecting a version of eslint that is ^7.23.0. The caret (^) in ^7.23.0 means any version that is 7.23.0 or higher, but less than 8.0.0.

To resolve this issue, you have a few options:

1. Downgrade your eslint to a version that is compatible with eslint-config-next. You can do this by running npm uninstall eslint followed by npm install eslint@7.32.0.

2. Upgrade eslint-config-next to a version that is compatible with eslint 8.33.0. You would need to check if such a version exists.

3. Use the --force or --legacy-peer-deps flag when running npm install as suggested in the error message. This will ignore the peer dependency conflict. However, this may lead to other issues in your project if eslint-config-next relies on features that are not available in eslint 8.33.0.

Remember to always backup your project or use version control before making these kinds of changes, as they can potentially break your project.
